### DISCUSSION TITLE
What is going on with GitHub? Cannot clone a repo

### DISCUSSION THREAD

**Discussion by Polda18** [community/community#23975] [New to GitHub]
<p>Hi. I’m, trying to clone my repositories over SSH, over HTTPS (I did create SSH key and used it in my account, I did create HTTPS token, and I saved it on my machine to be used), but both of them just time out?</p>
<pre><code class="lang-auto">$ git clone https://github.com/Polda18/CPU-6502-Emulator.git &lt; git_https_token.txt
Cloning into 'CPU-6502-Emulator'...
fatal: unable to access 'https://github.com/Polda18/CPU-6502-Emulator.git/': Failed to connect to github.com port 443: Connection timed out
</code></pre>
<pre><code class="lang-auto">$ git clone git@github.com:Polda18/CPU-6502-Emulator.git
Cloning into 'CPU-6502-Emulator'...
ssh: connect to host github.com port 22: Connection timed out
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
</code></pre>
<p>Repository does exist: <a href="https://github.com/Polda18/CPU-6502-Emulator" class="inline-onebox" rel="noopener nofollow ugc">GitHub - Polda18/CPU-6502-Emulator: Practice run repository for a 6502 CPU emulator</a></p>

**Comment by byrneh** [comment]
<aside class="quote no-group" data-username="polda18" data-post="1" data-topic="167513">
<div class="title">
<div class="quote-controls"></div>
 polda18:</div>
<blockquote>
<p><code>git clone https://github.com/Polda18/CPU-6502-Emulator.git</code></p>
</blockquote>
</aside>
<p>Have you tested your SSH access yet?<br>
$ ssh -T <a href="mailto:git@github.com">git@github.com</a><br>
You may see a warning like this:</p>
<blockquote>
<p>The authenticity of host ‘<a href="http://github.com" rel="noopener nofollow ugc">github.com</a> (IP ADDRESS)’ can’t be established.<br>
RSA key fingerprint is SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8.<br>
Are you sure you want to continue connecting (yes/no)?<br>
Hi username! You’ve successfully authenticated, but GitHub does not<br>
provide shell access.</p>
</blockquote>
<p>For HTTPS try only<br>
$ git clone <a href="https://github.com/Polda18/CPU-6502-Emulator.git" rel="noopener nofollow ugc">https://github.com/Polda18/CPU-6502-Emulator.git</a></p>

**Comment by Polda18** [comment]
<p>That times out as well :frowning:</p>

**Comment by byrneh** [comment]
<p>Do both the SSH test and git cone over HTTPS timeout?<br>
Are you connected via a proxy or direct to the internet?</p>
<p>If you run the following at the command line<br>
$ curl --location <a href="http://github.com/status" rel="noopener nofollow ugc">github.com/status</a><br>
Do you get a message as below returned<br>
GitHub lives! (2021-03-12 08:18:02 -0800) (1)</p>

**Reply by Array-ray** [reply]
oh my god, its amazing! I 'v tried several times cloning the asset from stable diffusion webui asset repo, but failed. And I tried this "curl --location [github.com/status](https://github.com/status)". I tried again after I got a lives messages, then it succeeded! Thank you.

**Comment by Polda18** [comment]
<p>Directly to the internet. I do use VPN, but I tried it with VPN disabled, and still the same result.</p>
<pre><code class="lang-auto">$ curl --location github.com/status
curl: (28) Failed to connect to github.com port 80: Connection timed out
</code></pre>
<p>I’m not connected to VPN right now, directly to Internet. Internet is working completely normal for me, but GitHub in console is completely timed out. SSH, HTTPS, nothing works. But it does work in browser. Strange.</p>

**Comment by byrneh** [comment]
<p>So not a GitHub issue but a configuration problem with your local device and/or Git client installation.<br>
Could also be related to some local firewall blocking from command line.</p>
<p>Can’t really guess what unless you provide steps to recreate including what OS, what Git client, what Git shell you are using etc.</p>
<p>Also from your Git Bash shell you can try<br>
$ openssl s_client -connect <a href="http://github.com:443" rel="noopener nofollow ugc">github.com:443</a><br>
to see what it says</p>
<p>If you are using Windows from PowerShell prompt try or MacOS commandline (not Git Bash or Git CMD)<br>
curl <a href="http://github.com/status" rel="noopener nofollow ugc">github.com/status</a><br>
it may narrow it down a little further and determine if it is limited to Git or wider command line issue.</p>

**Comment by Polda18** [comment]
<p>It may be a firewall issue. Nothing seems to work and I don’t quite know where I can allow GitHub SSH or HTTPS. I’m using Avast Antivirus software (Premium subscription), which also manages firewall.</p>
<pre><code class="lang-auto">$ openssl s_client -connect github.com:443
139900144690496:error:0200206E:system library:connect:Connection timed out:../crypto/bio/b_sock2.c:110:
139900144690496:error:2008A067:BIO routines:BIO_connect:connect error:../crypto/bio/b_sock2.c:111:
connect:errno=110
$ curl github.com/status
curl: (28) Failed to connect to github.com port 80: Connection timed out
</code></pre>
<p>I’m using WSL2 with Ubuntu on Windows 10. Do I need to use <code>git</code> on Windows instead?</p>

**Comment by Polda18** [comment]
<p>I can’t even use <code>wget</code> to download a file from GitHub Pages. What is going on?</p>
<pre><code class="lang-auto">$ wget https://github.7biom.cz/json/schemas/Notch/locale.json
--2021-03-13 10:55:00--  https://github.7biom.cz/json/schemas/Notch/locale.json
Resolving github.7biom.cz (github.7biom.cz)... 104.21.87.100, 172.67.169.52, 2606:4700:3033::6815:5764, ...
Connecting to github.7biom.cz (github.7biom.cz)|104.21.87.100|:443... failed: Connection timed out.
Connecting to github.7biom.cz (github.7biom.cz)|172.67.169.52|:443... ^C
</code></pre>
<p>The file does exist, BTW: <a href="https://github.7biom.cz/json/schemas/Notch/locale.json" rel="noopener nofollow ugc">https://github.7biom.cz/json/schemas/Notch/locale.json</a></p>

**Comment by Polda18** [comment]
<p>Okay, so I guess the culprit is WSL2 Ubuntu Bash, which seems to lack Internet connection. It does resolve IP address though, which wouldn’t be possible if it didn’t have Internet connection. For some reason, Bash is blocking any kind of download or upload. How do I fix this?</p>
<pre><code class="lang-auto">$ wget http://czghost.4fan.cz/ftp/downloads/quake3/maps/czg_sude.zip
--2021-03-13 11:01:05--  http://czghost.4fan.cz/ftp/downloads/quake3/maps/czg_sude.zip
Resolving czghost.4fan.cz (czghost.4fan.cz)... 88.86.121.75, 2a01:28:ca:63:121:75::
Connecting to czghost.4fan.cz (czghost.4fan.cz)|88.86.121.75|:80... failed: Connection timed out.
Connecting to czghost.4fan.cz (czghost.4fan.cz)|2a01:28:ca:63:121:75::|:80... failed: Network is unreachable.
</code></pre>
<p>The file I’m trying to download is here: <a href="http://czghost.4fan.cz/ftp/downloads/quake3/maps/" class="inline-onebox" rel="noopener nofollow ugc">Index of /ftp/downloads/quake3/maps</a><br>
These are my personal websites, nothing to do with GitHub, yet it seems that I’m having the same issue here as well…</p>

**Comment by byrneh** [comment]
<p>Yes that all conforms its a problem with your device/WSL set-up<br>
Although it should be possible to get this traffic allowed from there also.</p>

**Comment by Polda18** [comment]
<aside class="quote no-group" data-username="byrneh" data-post="10" data-topic="167513">
<div class="title">
<div class="quote-controls"></div>
 byrneh:</div>
<blockquote>
<p>Although it should be possible to get this traffic allowed from there also.</p>
</blockquote>
</aside>
<p>And how do I do that? I’m sorry, I’m not much familiar with Unix shell, I’m still learning.</p>

**Comment by byrneh** [comment, marked as answer]
<p>Hi <a class="mention" href="/polda18">@polda18</a><br>
ok I checked on my Windows 10 system (with default Windows virus/firewall configured) and ensuring I have the latest WSL2 installed and a fresh install of Ubuntu from Microsoft Store.<br>
Everything worked ok, including internet access after install with no further configuration needed.</p>
<p>From Windows Terminal<br>
wsl -l -v<br>
NAME      STATE           VERSION<br>
* Ubuntu    Running         2</p>
<p>From Ubuntu distro Shell<br>
<span class="math"> uname -r
5.4.72-microsoft-standard-WSL2
</span> cat /etc/issue<br>
Ubuntu 20.04.2 LTS \n \l<br>
$ curl --location <a href="http://github.com/status" rel="noopener nofollow ugc">github.com/status</a><br>
GitHub lives! (2021-03-13 18:13:10 -0800) (1)</p>
<p><a href="https://docs.microsoft.com/en-us/windows/wsl/faq#why-do-i-not-have-internet-access-from-wsl" rel="noopener nofollow ugc">WSL FAQs why-do-i-not-have-internet-access-from-wsl</a><br>
Some users have reported issues with specific firewall applications blocking internet access in WSL. The firewalls reported are:</p>
<ol>
<li>Kaspersky</li>
<li>AVG</li>
<li><strong>Avast</strong></li>
</ol>
<p>In some cases turning off the firewall allows for access. In some cases simply having the firewall installed looks to block access.<br>
If you want to try and get working with your curr3nt configuration, yo could try google search related to WSL, Avast and internet access blocker issues</p>

**Comment by Polda18** [comment]
<p>Yes, it was Firewall. I simply switched the network profile to private/home (because for some reason it has been set to public), and now it works :smiley:</p>

**Comment by ghost** [comment]
<p>I also had the same problem and turning off my firewall helped me and I was using quick heal</p>

**Comment by Polda18** [comment]
<p>I wouldn’t recommend anyone to turn off their firewall, maybe just temporarily to see if it is caused by firewall, but I would recommend changing firewall settings for sure. How? That depends on which firewall you’re using.</p>

**Comment by chucklu** [comment]
<p>It’s none business of GitHub.  You need ask help from your network service provider.<br>
You might check the network problem via <code>ping</code> and <code>tracert</code> command.<br>
ping <a href="http://github.com" rel="noopener nofollow ugc">github.com</a> -t<br>
tracert -d <a href="http://github.com" rel="noopener nofollow ugc">github.com</a></p>

**Comment by franque01** [comment]
It worked, I changed my firewall's network to private
